swift/lib/dependabot/swift/metadata_finder.rb

```diff
@@ -1,14 +1,18 @@
-# typed: true
+# typed: strict
 # frozen_string_literal: true
 
 require "dependabot/metadata_finders"
 require "dependabot/metadata_finders/base"
+require "sorbet-runtime"
 
 module Dependabot
   module Swift
     class MetadataFinder < Dependabot::MetadataFinders::Base
+      extend T::Sig
+
       private
 
+      sig { override.returns(T.nilable(T.any(Dependabot::Source, T.untyped))) }
```

Comment:
`T.any(Dependabot::Source, T.untyped)` is equivalent to `T.untyped`. I don't think you need the union with `T.untyped` here?

```suggestion
      sig { override.returns(T.nilable(Dependabot::Source)) }
```

Reply:
@raj-meka I spoke with you offline about this.

The error is due to `find_source_from_registry` returning `void`. I thought this method came from `Dependabot::MetadataFinders::Base`, but it doesn't. It looks like support for `registry` source types was never implemented.

@deivid-rodriguez I was wondering if you remember the intention here?

If not, I'd suggest changing the return type of `find_source_from_registry` to `returns(T.noreturn)` works ([See the documentation here][1]).

[1]: https://sorbet.org/docs/noreturn#declaring-that-a-method-always-raises

Reply:
I have done change as per review comment

Reply:
I left it unimplemented, yes. When we first added Swift support, registry source support had been implemented in SwiftPM, but most people seemed to not be using it and relying on git sources instead. So I didn't tackle registry support initially. It could be implemented if registry sources have gained any traction in Swift.